src/EvoSC.Common/Exceptions/EvoSCException.cs

```diff
@@ -0,0 +1,22 @@
+﻿using System.Runtime.Serialization;
```

Comment:
Is it neccessary with a EvoSC prefixed exception? Wouldn't all exceptions thrown, be a evosc exception?

Reply:
The convention is to create a base exception for your application that all other exceptions inherit. It makes it easier to separate what is coming from the app and other places like libs. I prefixes the exceptions to avoid naming conflicts.

Reply:
I think the prefixed Exceptions here are fine based on snix's argumentation.